stats/thresholds_test.go

```diff
@@ -25,99 +25,250 @@ import (
+}
+
+func TestThreshold_runNoTaint(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		Source           string
+		LastFailed       bool
+		AbortOnFail      bool
+		AbortGracePeriod types.NullDuration
+		parsed           *thresholdExpression
+	}
+	type args struct {
+		sinks map[string]float64
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			"valid expression using the > operator over passing threshold",
+			fields{"rate>0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenGreater, 0.01}},
+			args{map[string]float64{"rate": 1}},
+			true,
+			false,
+		},
+		{
+			"valid expression using the >= operator over passing threshold",
+			fields{"rate>=0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenGreaterEqual, 0.01}},
+			args{map[string]float64{"rate": 0.01}},
+			true,
+			false,
+		},
+		{
+			"valid expression using the <= operator over passing threshold",
+			fields{"rate<=0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenLessEqual, 0.01}},
+			args{map[string]float64{"rate": 0.01}},
+			true,
+			false,
+		},
+		{
+			"valid expression using the < operator over passing threshold",
+			fields{"rate<0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenLess, 0.01}},
+			args{map[string]float64{"rate": 0.00001}},
+			true,
+			false,
+		},
+		{
+			"valid expression using the == operator over passing threshold",
+			fields{"rate==0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenLooselyEqual, 0.01}},
+			args{map[string]float64{"rate": 0.01}},
+			true,
+			false,
+		},
+		{
+			"valid expression using the === operator over passing threshold",
+			fields{"rate===0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenStrictlyEqual, 0.01}},
+			args{map[string]float64{"rate": 0.01}},
+			true,
+			false,
+		},
+		{
+			"valid expression using != operator over passing threshold",
+			fields{"rate!=0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenBangEqual, 0.01}},
+			args{map[string]float64{"rate": 0.02}},
+			true,
+			false,
+		},
+		{
+			"valid expression over failing threshold",
+			fields{"rate>0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenGreater, 0.01}},
+			args{map[string]float64{"rate": 0.00001}},
+			false,
+			false,
+		},
+		{
+			"valid expression over non-existing sink",
+			fields{"rate>0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenGreater, 0.01}},
+			args{map[string]float64{"med": 27.2}},
+			false,
+			true,
```

Comment:
I definitely want to know what the fifth element here is without having to go up to the definition of the type.

Reply:
Agreed 👍🏻